### PR TITLE
Function renderWord complete

### PR DESCRIPTION
**Pull request by mattlusty**
As its difficult to see what I changed (due to our different auto-format-on-save) I'll try and summarise the areas I changed that were outside of my allocated area (the renderWord function), within the below comments.

Otherwise I just added my renderWord function, and it seems to work.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,100 +1,113 @@
...
+            repellat voluptatem amet, nostrum soluta rem quis voluptate. Neque necessitatibus
+            quisquam, officiis, sint quod quaerat, doloribus velit at quam rem hic id eligendi cum
+            non.
+          </p>
+        </section>
 
-      <section id="pronunciation" class="my-1">
-        <h2>Pronunciation</h2>
-        <audio controls>
-          <source src="horse.mp3" type="audio/mpeg" />
```
Comment (mattlusty) [inline index.html:42]: @Whefert - HTML changes

This is the only HTML I believe I touched. (Perhaps next time, I should have done this in its own separate commit and Pull Request)

Line 42. I removed the "horse.mp3" from the audio source element's src attribute (as it was causing a 404 error in the console when trying to fetch it). I presume it was only a placeholder anyway right?

File: assets/js/script.js [hunk 2]
```diff
@@ -1,54 +1,111 @@
 var searchInputText = "Hello";
 var searchInputElement = $("#search");
-var wordDefinitionArea = $(".definition");
-var wordPronunciationArea = $("#audio");
+var wordDefinitionArea = $("#definition");
+var wordPronunciationArea = $("#pronunciation");
```
Comment (mattlusty) [inline assets/js/script.js:4]: @HillaryM17 - JS Changes related to your global variables

I just changed the jQuery selectors (and thus the selected elements) on these 2 lines.

1) The #definition element uses ID (instead of a class)
2) There was no element with #audio ID so I changed to #pronunciation

I needed to access these elements, hence my meddling here!

File: assets/js/script.js [hunk 3]
```diff
@@ -1,54 +1,111 @@
...
+    .then(function (data) {
+      console.log("words data: ", data);
+    });
+
+  fetch(speechURL, speechOptions).then(function (response) {
+    console.log(response);
+  });
+}
+
+// onsearch();
```
Comment (mattlusty) [inline assets/js/script.js:50]: @HillaryM17 

I just commented out where you calling onsearch(). This stopped the console.logs within your function getting in the way. This was/is just for your testing purposes I presumed.

I've actually left a similar redundant test call to my function renderWord() as well. (However there are no console.logs within this function, so should not interfere fingers crossed).

**Comment by HillaryM17** [comment]
Hey, looks good :) 
Is it worth declaring the variables you have for the data within the onSearch function? I guess it's tricky to know until we have event listeners and the functions work together. But happy to merge as is and can always tweak after :)

**Comment by HillaryM17** [comment]
Also, upon further investigation into the onSearch function and what the API produces, it can offer multiple definitions also... I think it's worth doing a forEach for definition @Whefert we might need to amend the HTML to represent this also?

**Comment by mattlusty** [comment]
Could make it support multiple definitions to make it better.

I say from your onSearch function, do as you suggest and assume my renderWord(data) function takes an array (data.definitions).

In a follow up commit, later I will update my renderFunction(data) to expect an array under data.definitions.

(We can then decide whether to actually render all / some / one as an implementation detail with renderWord. (For now I'll pick the first out the array and render one, but we can discuss with Front End if and how to render more)

**Comment by mattlusty** [comment]
> Also, upon further investigation into the onSearch function and what the API produces, it can offer multiple definitions also... I think it's worth doing a forEach for definition @Whefert we might need to amend the HTML to represent this also?

Could make it support multiple definitions to make it better.

I say from your onSearch function, do as you suggest and assume my renderWord(data) function takes an array (data.definitions).

In a follow up commit, later I will update my renderFunction(data) to expect an array under data.definitions.

(We can then decide whether to actually render all / some / one as an implementation detail within renderWord. (For now I'll pick the first out the array and render one, but we can discuss with Front End if and how to render more)

**Comment by mattlusty** [comment]
> Hey, looks good :) 
> 
> Is it worth declaring the variables you have for the data within the onSearch function? I guess it's tricky to know until we have event listeners and the functions work together. But happy to merge as is and can always tweak after :) 

I think thats the idea. These data properties will all be declared within your onSearch function and then assigned with values based on the fetched response data. And the you can add them to a new object, and pass it into renderWord(data).

(So you're right, that data object I wrote in the code above (will be deleted, that was just a placeholder for when your onSearch function will be passing an actual data object.)

**Comment by mattlusty** [comment]
> Hey, looks good :) 
> 
> Is it worth declaring the variables you have for the data within the onSearch function? I guess it's tricky to know until we have event listeners and the functions work together. But happy to merge as is and can always tweak after :) 
> 
> 
> 

Eventually your onSearch function will be the source of this data object. So you will need to declare these variables within your onSearch function. 

Something along the lines of

fetch stuff, then ...

var word = (the actual word, I just thought this should definitely be included in this data that you pass renderWord! Although currently I dont think the html design includes the name of the word? That needs adding to HTML design @Whefert unless I'm mistaken?)

var definitions = (array pop with def strings Im assuming)

var audio = (whatever this is?)

var examples = (array of example strings Im assuming)

//then pack it into an object, you could call it whatever you like, but perhaps "data" or "wordData" or whatever...

var dataStuff = {
 word: word
 definitions: definitions,
 audio: audio,
 examples: examples
}

renderWord(dataStuff)

... onSearch code logic continues

**Comment by mattlusty** [comment]
All good points @HillaryM17 thanks.

We so if we tailor our code based on the above points. This should be good.

I'll make changes in a follow up commit. If you could push this for now.

Unless @Whefert has an issue with me removing the horse.mp3?

**Review by HillaryM17 — APPROVED** [review]
